components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java

```diff
@@ -1481,4 +1493,40 @@ private String removeCRLFCharacters(String str) {
         return str;
     }
 
+    private void loadAndSaveAnalyticsSolutions(String solutionType) throws TemplateManagerServiceException {
+        if (Objects.nonNull(availableTemplateGroups.get(solutionType))) {
+            TemplateGroup templateGroup = availableTemplateGroups.get(solutionType);
+            List<RuleTemplate> ruleTemplates = templateGroup.getRuleTemplates();
+
+            for (RuleTemplate ruleTemplate : ruleTemplates) {
+                BusinessRuleFromTemplate businessRuleFromTemplate = replacePropertiesWithDefaultValues
+                        (ruleTemplate, templateGroup.getUuid());
+                try {
+                    createTablesIfNotExist();
+                    BusinessRule businessRule = queryExecutor.retrieveBusinessRule(ruleTemplate.getUuid());
+
+                    if (businessRule == null) {
+                        saveBusinessRuleDefinition(ruleTemplate.getUuid(), businessRuleFromTemplate,
+                                TemplateManagerConstants.DEPLOYED, 1);
+                        updateArtifactCount(ruleTemplate.getUuid(), 1);
+                    }
+                } catch (BusinessRulesDatasourceException | TemplateInstanceCountViolationException e) {
+                    throw new TemplateManagerServiceException("Saving APIM analytics feature business rule '" +
```

Comment:
Please check whether we need to add "APIM analytics feature" here because of the template manger common to all the solution. In that case, maybe we can append the solution type here.